SliceTracker/SliceTracker.py

```diff
@@ -3576,3 +3596,150 @@ def onCaseNumberChanged(self, caseNumber):
     self.preview.setText("New case directory: " + self.newCaseDirectory)
     self.okButton.enabled = not os.path.exists(self.newCaseDirectory)
     self.notice.text = "" if not os.path.exists(self.newCaseDirectory) else "Note: Directory already exists."
+
+
+class TargetDisplacementChartWidget(object):
+    
+  @property
+  def chartView(self):
+    return self._chartView
+  
+  @property
+  def chart(self):
+    return self._chartView.chart()
+    
+  @property
+  def xAxis(self):
+    return self.chart.GetAxis(1)
+  
+  @property
+  def yAxis(self):
+    return self.chart.GetAxis(0)  
+
+  @property
+  def showLegend(self):
+    return self._showLegend
+
+  @showLegend.setter
+  def showLegend(self, value):
+    assert type(value) is bool, "Only boolean values are allowed for this class member"
+    self._showLegend = value
+    self.chart.SetShowLegend(value)
+
+  def __init__(self):
+    self._chartView = ctk.ctkVTKChartView()
+    self._chartView.minimumSize = qt.QSize(200,200)
+    self._showLegend = False
+    self.xAxis.SetTitle('Time Unit')
+    self.yAxis.SetTitle('Displacement')
+    self.chartTable = vtk.vtkTable()
+
+    self.arrX = vtk.vtkFloatArray()
+    self.arrX.SetName('X Axis')
+    
+    self.arrXD = vtk.vtkFloatArray()
+    self.arrXD.SetName('X Component')
+    
+    self.arrYD = vtk.vtkFloatArray()
+    self.arrYD.SetName('Y Component')
+    
+    self.arrZD = vtk.vtkFloatArray()
+    self.arrZD.SetName('Z Component')
+
+    self.arrX.InsertNextValue(0)
+    self.arrXD.InsertNextValue(0)
+    self.arrYD.InsertNextValue(0)
+    self.arrZD.InsertNextValue(0)
+
+    self.chartPopupWindow = None
+    self.chartPopupSize = qt.QSize(600, 300)
+    self.chartPopupPosition = qt.QPoint(0,0)
+    
+    self.showLegendCheckBox = qt.QCheckBox('Show legend')
+    self.showLegendCheckBox.setChecked(0)
+
+    self.plotFrame = ctk.ctkCollapsibleButton()
+    self.plotFrame.text = "Plotting"
+    self.plotFrame.collapsed = 0
+    plotFrameLayout = qt.QGridLayout(self.plotFrame)
+    self.plottingFrameWidget = qt.QWidget()
+    self.plottingFrameLayout = qt.QGridLayout()
+    self.plottingFrameWidget.setLayout(self.plottingFrameLayout)
+    self.plottingFrameLayout.addWidget(self.showLegendCheckBox,0,0)
+    self.plottingFrameLayout.addWidget(self._chartView,1,0)
+
+    self.popupChartButton = qt.QPushButton("Undock chart")
+    self.popupChartButton.setCheckable(True)
+    self.plottingFrameLayout.addWidget(self.popupChartButton,2,0)
+
+    self.DisplacementChartTargetSelector = qt.QComboBox()
+    self.targetSelectorLayout = qt.QFormLayout()
+    self.targetSelectorLayout.addRow("Target:",self.DisplacementChartTargetSelector)
```

Comment:
Just to be consistent, please use a space after `,` In some cases you are doing that...